### ISSUE TITLE
Duplication in roles list

### ISSUE THREAD

**Issue by brianshumate** [ansible/galaxy-issues#117]
I clicked on **Re-import Role** to being re-importing a role, switched tabs, and came back to the Galaxy tab where I observed that the role list was now duplicating every role:

![duped_roles](https://cloud.githubusercontent.com/assets/77563/12875743/c41ec0ca-cdbe-11e5-8a52-bc4c1ef458bd.png)

The list remains in this state despite reloading or even shift-reloading the page. I have also signed out and back into Galaxy, but the list view continues to show duplicate role entries.

**Comment by chouseknecht** [comment]
I am able to create duplicate repos in the cache for my user. I did it by clicking the Refresh button, going to a different tab before the refresh finished, and then coming back to My Roles and clicking the Refresh button again. Not sure that this is the same workflow, but it does point out an obvious problem that may be the root cause- there should be a unique key of owner, github_user, github_repo on the cache. 

I've added the unique key index and adjusted the cache refresh process to handle duplicate key exceptions. This is now available for testing at galaxy-qa.ansible.com.
